From b88ebeb328d57894a19330ddf0ef021d23f4066f Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Fri, 6 Sep 2013 19:08:36 -0700
Subject: [PATCH 10/10] OMAP3: overo: Work around HW ECC error.

This is a workaround to disable subpage reads when the
ECC mode is not NAND_ECC_SOFT as discussed:
http://u-boot.10912.n7.nabble.com/NAND-write-error-with-HW-ECC-on-OMAP3-td162420.html

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 drivers/mtd/nand/nand_base.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/nand/nand_base.c b/drivers/mtd/nand/nand_base.c
index 9e05cef..7c4cd45 100644
--- a/drivers/mtd/nand/nand_base.c
+++ b/drivers/mtd/nand/nand_base.c
@@ -3344,6 +3344,8 @@ int nand_scan_tail(struct mtd_info *mtd)
 	/* Large page NAND with SOFT_ECC should support subpage reads */
 	if ((chip->ecc.mode == NAND_ECC_SOFT) && (chip->page_shift > 9))
 		chip->options |= NAND_SUBPAGE_READ;
+	else
+		chip->options &= ~NAND_SUBPAGE_READ;
 
 	/* Fill in remaining MTD driver data */
 	mtd->type = MTD_NANDFLASH;
-- 
1.7.10.4

